oak_restricted_kernel/src/lib.rs

```diff
@@ -285,7 +285,19 @@ pub fn start_kernel(info: &BootParams) -> ! {
     let application = payload::Application::load_raw(&mut *channel)
         .expect("failed to load application binary from channel");
 
-    syscall::enable_syscalls(channel);
+    let derived_key = if sev_snp_enabled {
+        snp_guest::get_derived_key().expect("couldn't derive key")
+    } else {
+        // Zero fixed key since we can't derive a consistent key without SEV-SNP.
+        DerivedKey::default()
+    };
+    // Mix the application digest into the final derived key.
+    let mut extraction = HkdfExtract::<Sha256>::new(Some(b"Oak application key"));
```

Comment:
What's the `"Oak application key"`?

Reply:
That is just an initial SALT needed by the HKDF. This can be anything, including and empty string so I chose a string that indicates the purpose of the key derivation.